pkg/analysis/statussubresource/analyzer.go

```diff
@@ -0,0 +1,163 @@
+package statussubresource
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/token"
+
+	"github.com/JoelSpeed/kal/pkg/analysis/helpers/extractjsontags"
+	"github.com/JoelSpeed/kal/pkg/analysis/helpers/markers"
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+const (
+	name = "statussubresource"
+
+	statusJSONTag = "status"
+
+	// kubebuilderRootMarker is the marker that indicates that a struct is the object root for code and CRD generation.
+	kubebuilderRootMarker = "kubebuilder:object:root:=true"
```

Comment:
So, testing this against o/api, we have `kubebuilder:object:root=true` and not `kubebuilder:object:root:=true`. Can we check if the `:=` version is supported with Kubebuilder and then work out a way to have this work

I wonder if we need to parse the markers slightly differently, so that `:=` and `=` differences are handled the same as in Kubebuilder (though I think they aren't consistent)

Reply:
According to https://book.kubebuilder.io/reference/markers/object the "correct" way is with `:=`. I'm happy to update this such that it also accepts `=`.

I know it is slated in the desired linters for the future, but that marker typos linter would probably be helpful here in fixing these types of issues and allowing other linters to only consider markers as are documented.

I'm happy to take a stab at an initial implementation of that linter that could be expanded in the future.

In the meantime, I'll explore ways to improve this logic such that it is `:=` or `=` agnostic.

Reply:
Agreed, settling on `:=` is probably what we want to do, I've been exploring the logic, and basically, it falls back to [here](https://github.com/kubernetes-sigs/controller-tools/blob/425b7d770a963e4b4cdcd389b85c8501287a97f3/pkg/markers/reg.go#L143-L154) where name and anonName are either `kubebuilder:object:root` and `kubebuilder:object:root:` or `kubebuilder:object` and `kubebuilder:object:root` depending on if you have `:=` or just `=`

There are a few things we could do:
* Just handle both in this linter for now - spread the problem out but take the easy path
* Enhance the marker extraction to return markers based on their name, and provide multiple values - larger, but better long term, can encapsulate marker discrepancies all within that package
* Try and enforce `:=` everywhere - Not sure if that is compatible with upstream markers like `+default` or not, I think that is `+default=` and doesn't work with the `:=`

I'm tempted to look at re-working the marker extraction so that we look up marker names, and they then optional have a list of values.

Or else, we have to create a marker registry and import the full marker parsing from controller-tools, may be a little heavy handed?